File: backend/lib/transferFunction.ts
export class TransferFunction {
    
    private noColours: number;
    private colourFormat: string;
    private filename: string;

    constructor(meta: any[]) {
        this.noColours = Number(meta['Size']['_']);
        this.filename = meta['Filename']['_']; 
        
        this.findFormat(meta['Pixel_Format']['_']);
    }

    private findFormat(format: string) {
        if (format == 'rgba32f') this.colourFormat = 'rgba32float';
        else console.error('Invalid pixel format for transfer function texture.');
    }

    public getFilename(): string { return this.filename; }

}

File: src/transfer_function.py
class TransferFunction:

    def __init__(self, meta):
        self.noColours = int(meta['Size'])
        self.filename = meta['Filename']
        self.find_format(meta['Pixel_Format'])

    def find_format(self, pixel_format):
        if pixel_format == 'rgba32f': self.colourFormat = 'rgba32float'
        else: raise Exception('Invalid pixel format for transfer function texture.')